app/assets/javascripts/modules/remove-filter.js

```diff
@@ -57,10 +52,9 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
         var haystack = ' ' + currentVal.trim() + ' '
         var needle = ' ' + decodeEntities(removeFilterValue.toString()) + ' '
         var newVal = haystack.replace(needle, ' ').replace(/ {2}/g, ' ').trim()
-
-        $input.val(newVal).trigger(onChangeSuppressAnalytics)
+        window.GOVUK.triggerEvent($input.val(newVal)[0], 'change', { detail: { suppressAnalytics: true } })
```

Comment:
JQuery spotted❗ 👀 ❗ I'm trying to understand what's happening here and struggling. The way jQuery applies new input values is using a function call instead of the declaration that plain js uses (so `$input.value = newVal` instead of `$input.val(newVal)`). It looks like we're passing the output of the jQuery input value update function to the trigger event? This feels a bit grubby, and also looks like we're hiding the update of an input value in here. If there's an explanation I'm all ears, otherwise **get that jQuery out of here**.

Reply:
Hi Owen, basically this PR is only concerned with the expander. The changes made here to the remove-filter are the absolute bare minimum that this needs to pass. I've got [a branch](https://github.com/alphagov/finder-frontend/compare/remove-filter) that's based on this one that's going to properly address all of the jQuery in this file separately. I promise all of the jQuery is going :D

Reply:
I seeeeeeee. Cool! Will drop a tick on this now.